### ISSUE TITLE
Double accounting for semantic_loss in total_loss and hence self_loss

### ISSUE THREAD

**Issue by seamie6** [GANWANSHUI/GaussianOcc#35]
Hello here is the code snippet I am referring to in ```runner.py```, for some reason I cannot reference it from the source code

```python
    semantic_loss = self.opt.semantic_loss_weight * semantic_loss
    losses[f"loss_semantic/{scale}"] = semantic_loss


loss_reg = 0
for k, v in outputs.items():
    if isinstance(k, tuple) and k[0].startswith("loss") and k[1] == scale:
        losses[f"{k[0]}/{k[1]}"] = v
        loss_reg += v

total_loss += loss + loss_reg + semantic_loss
```

Here ```total_loss``` is a combination of three losses, most notably it includes ```semantic_loss```and ```loss```
```loss``` is a combination of losses from the ```losses``` dictionary, which includes ```semantic_loss``` if I am not mistaken as it seems to be stored in it, hence ```semantic_loss``` is doubly accounted for in ```total_loss``` and hence ```self_loss```. Is this intentional?

**Comment by seamie6** [comment]
It was a misunderstanding, it is looping over the losses stored in outputs.items(), not the losses dictionary. Hence no double accounting

**Comment by seamie6** [comment]
However, there is a double accounting for loss if using ```get_semantic_gt_loss()``` inside of ```occupancy_decoder.py``` for ```loss_geometry```:

```python
  self.outputs[("loss_geometry", 0)] = loss_geometry
  self.outputs[("loss_gt_occ", 0)] = loss_occ * 100 + loss_geometry
```

Since both are stored in the ```outputs``` dictionary they will be summed together for ```total_loss```. This is not used for the self-supervised model though.
